.github/workflows/ci.yml

```diff
@@ -1,45 +1,149 @@
-name: Node CI
+name: CI
 
 on:
   push:
     branches:
       - main
-    tags:
-      - 'v*'
-  pull_request:
+      - master
+  pull_request: {}
+
+concurrency:
+  group: ci-${{ github.head_ref || github.ref }}
+  cancel-in-progress: true
 
 jobs:
-  lint:
+  test:
+    name: "Tests"
     runs-on: ubuntu-latest
-    outputs:
-      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    timeout-minutes: 10
+
     steps:
       - uses: actions/checkout@v4
-      - uses: wyvox/action-setup-pnpm@v3
```

Comment:
if you use this, you don't need to do all of
```
  - uses: pnpm/action-setup@v3
        with:
          version: 9
      - uses: actions/setup-node@v4
        with:
          node-version: 18
          cache: pnpm
      - name: Install Dependencies
        run: pnpm install --frozen-lockfile
```

Reply:
done